src/routers/handlers/yourCompanies/companyInvitationsHandler.ts

```diff
@@ -0,0 +1,61 @@
+import { Request } from "express";
+import { GenericHandler } from "../genericHandler";
+import { getTranslationsForView } from "../../../lib/utils/translations";
+import * as constants from "../../../constants";
+import { getLoggedInUserEmail } from "../../../lib/utils/sessionUtils";
+import { Association, AssociationStatus, Associations } from "../../../types/associations";
+import { getUserAssociations } from "../../../services/userCompanyAssociationService";
+import { AnyRecord, ViewData } from "../../../types/util-types";
+import { getUserRecord } from "../../../services/userService";
+
+export class CompanyInvitationsHandler extends GenericHandler {
+    async execute (req: Request): Promise<ViewData> {
+        this.viewData = await this.getViewData(req);
+        return Promise.resolve(this.viewData);
+    }
+
+    private async getViewData (req: Request): Promise<ViewData> {
+        const userEmailAddress = getLoggedInUserEmail(req.session);
+        const userAssociations: Associations = await getUserAssociations(userEmailAddress);
+        const awaitingApproval = userAssociations.items?.filter(item => item.status === AssociationStatus.AWAITING_APPROVAL);
```

Comment:
the new endpoint we have will take in query parameters for status

Reply:
All the code that is directly involved in API calls will be redone once all the APIs are available via api-sdk-node. There is no point making any changes before then.